File: python/day-1a.py
with open('input/day-1.txt', 'r') as f:
    lines = f.read().split('\n')
    line_nums = [[int(c) for c in l if c.isdigit()] for l in lines]
    print(sum(l[0] * 10 + l[-1] for l in line_nums))

File: rust/src/bin/day-1a.rs
fn main() -> std::io::Result<()> {
    let total: u32 = include_str!("../../../input/day-1.txt")
        .split("\n")
        .map(|x: &str| x.chars().filter(|x: &char| x.is_ascii_digit()).collect())
        .map(|x: Vec<char>| (x[0] as u32 - 48) * 10 + (*x.last().unwrap() as u32 - 48))
        .sum::<u32>();

    println!("{}", total);
    Ok(())
}
